node_cli/mirage/record/redis_record.py

```diff
@@ -0,0 +1,151 @@
+#   -*- coding: utf-8 -*-
+#
+#   This file is part of node-cli
+#
+#   Copyright (C) 2025-Present SKALE Labs
+#
+#   This program is free software: you can redistribute it and/or modify
+#   it under the terms of the GNU Affero General Public License as published by
+#   the Free Software Foundation, either version 3 of the License, or
+#   (at your option) any later version.
+#
+#   This program is distributed in the hope that it will be useful,
+#   but WITHOUT ANY WARRANTY; without even the implied warranty of
+#   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+#   GNU Affero General Public License for more details.
+#
+#   You should have received a copy of the GNU Affero General Public License
+#   along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+import abc
+from typing import Any
+from datetime import datetime
+from dataclasses import dataclass
+
+import redis
+
+from node_cli.configs import REDIS_URI
+
+cpool: redis.ConnectionPool = redis.ConnectionPool.from_url(REDIS_URI)
+rs: redis.Redis = redis.Redis(connection_pool=cpool)
+
+@dataclass
+class FieldInfo:
+    name: str
+    type: type
+    default: str | int | bool | datetime | None
+
+
+class FlatRedisRecord:
+    def __init__(self, name: str):
+        self.name = name
+        if not self._exists():
+            self._set_defaults()
+            self._save()
+
+    def to_dict(self) -> dict:
+        return self.mget(*self._record_fields().keys())
+
+    def mget(self, *args) -> dict[str, Any]:
+        key_names = [self._get_field_key(field_name) for field_name in args]
+        raw_res = rs.mget(*key_names)
+        return {
+            key_name: self._deserialize_field(value, self._record_fields()[key_name].type)
+            for value, key_name in zip(raw_res, args)
+        }
+
+    def mset(self, **kwargs) -> None:
+        key_names = [self._get_field_key(field_name) for field_name in kwargs.keys()]
+        values = [
+            self._serialize_field(value, self._record_fields()[field_name].type)
+            for value, field_name in zip(kwargs.values(), kwargs.keys())
+        ]
+        rs.mset(dict(zip(key_names, values)))
+
+    def delete(self) -> None:
+        rs.delete(*self._key_names())
+
+    def _get_field_key(self, field_name: str) -> str:
+        return f'{self.name}_{field_name}'
+
+    def _serialize_datetime(self, dt: datetime) -> str:
+        return dt.isoformat()
+
+    def _deserialize_datetime(self, value: str) -> datetime:
+        return datetime.fromisoformat(value)
+
+    def _get_field(self, field_name: str):
+        key = self._get_field_key(field_name)
+        value = rs.get(key)
+        if value is None:
+            raise ValueError(f"Field '{field_name}' not found in record '{self.name}'")
```

Comment:
this line will cause a crash if you try to access the variable with default value = None

suggesting to change to:
```
    def _get_field(self, field_name: str):
        key = self._get_field_key(field_name)
        value = rs.get(key)
        return self._deserialize_field(value, self._record_fields()[field_name].type)
``` 
(same as admin)

None result is already handled in `_deserialize_field`